### PR TITLE
chore: backport to v0.34.x tendermint

### LINKED ISSUE #951: chore: prepare node for backport to v0.34.x tendermint
In order to be aligned with efforts made in app/core on downgrades, node should also do downgrade of respective deps 

Ref: 
- https://github.com/celestiaorg/celestia-core/pull/802
- https://github.com/celestiaorg/celestia-core/issues/790

### PR DESCRIPTION
**Pull request by distractedm1nd**
Resolves #951

### CODE REVIEW

**Comment by distractedm1nd** [comment]
@renaynay The main issue currently seems to be with how the config works in `core/testing.go` - haven't been able to figure it out yet though

**Comment by distractedm1nd** [comment]
@renaynay There are now only 4 tests that are failing, but I can't figure out why yet. 3 of them pass when run individually, but get a context timeout when run with the others for some reason. The other one, `TestFraudProofBroadcasting`, has some file write issue that I haven't been able to track down yet. Will keep looking, but if you could checkout and take a look it might be something stupid easy that I'm just overseeing

**Review by Bidon15 — COMMENTED** [review]
imho, we should wait until app/core cut proper releases to merge this awesome PR 
Thanks for the heavy-lifting @distractedm1nd 💪

File: go.mod [hunk 1]
```diff
@@ -6,10 +6,10 @@ replace github.com/ipfs/go-verifcid => github.com/celestiaorg/go-verifcid v0.0.1
 
 require (
 	github.com/BurntSushi/toml v1.2.0
-	github.com/celestiaorg/celestia-app v0.5.4
+	github.com/celestiaorg/celestia-app v0.6.1-0.20220802211248-87653d6cb78e
```
[the remark this answers is not in the document]
Comment (distractedm1nd) [reply go.mod:9]: Definitely, same with core, waiting for the release to update

[the remark this answers is not in the document]
Comment (evan-forbes) [reply go.mod:9]: I'll go create release candidates, one sec

File: core/testing.go [hunk 2]
```diff
@@ -29,7 +29,11 @@ func StartTestNode(ctx context.Context, t *testing.T, app types.Application, cfg
 		options.SpecificConfig = cfg
 	})
 	t.Cleanup(func() {
-		rpctest.StopTendermint(nd)
+		err := nd.Stop()
```
Comment (renaynay) [inline core/testing.go]: @distractedm1nd can you doc why we do this?

File: go.mod [hunk 3]
```diff
@@ -249,30 +248,29 @@ require (
...
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace (
-	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v1.1.0-sdk-v0.46.0
+	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v1.2.0-sdk-v0.46.0
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 	github.com/libp2p/go-libp2p-pubsub v0.7.0 => github.com/celestiaorg/go-libp2p-pubsub v0.6.2-0.20220628100036-657948473f1f
-	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.2.4-tm-v0.35.6
+	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.3.0-tm-v0.34.20.0.20220805165832-5401107e9616
```
[the remark this answers is not in the document]
Comment (liamsi) [reply go.mod:275]: https://github.com/celestiaorg/celestia-core/releases/tag/v1.3.1-tm-v0.34.20

**Review by liamsi — CHANGES_REQUESTED** [review]
Use the latest core release. Otherwise looks good!

File: core/testing.go (hunk 2 above)
Comment (distractedm1nd) [reply core/testing.go]: Hey, think thats from an older commit, `rpctest.StopTendermint` works again :D !

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (distractedm1nd) [reply go.mod]: done

File: go.mod (hunk 3 above)
[the remark this answers is not in the document]
Comment (distractedm1nd) [reply go.mod]: done

**Review by liamsi — APPROVED** [review]

File: node/tests/swamp/swamp.go [hunk 4]
```diff
@@ -69,6 +69,10 @@ func NewSwamp(t *testing.T, options ...Option) *Swamp {
 	require.NoError(t, err)
 	remote, err := core.NewRemote(ip, port)
 	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := remote.Stop()
+		require.NoError(t, err)
+	})
```
Comment (renaynay) [inline node/tests/swamp/swamp.go:75]:
```suggestion
```

**Review by renaynay — COMMENTED** [review]
* downgrade rsmt2d to 0.5.0 (as core has it)
* test against backported core

File: core/testing.go (hunk 2 above)
Comment (renaynay) [reply core/testing.go]: shipit

**Comment by renaynay** [comment]
We need to rebase :(

File: go.mod [hunk 5]
```diff
@@ -263,30 +263,29 @@ require (
...
 	lukechampine.com/blake3 v1.1.7 // indirect
-	nhooyr.io/websocket v1.8.6 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace (
-	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v1.1.0-sdk-v0.46.0
+	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v1.2.0-sdk-v0.46.0
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 	github.com/libp2p/go-libp2p-pubsub v0.7.0 => github.com/celestiaorg/go-libp2p-pubsub v0.6.2-0.20220628100036-657948473f1f
```
Comment (Bidon15) [inline go.mod:289]:
```suggestion
	github.com/libp2p/go-libp2p-pubsub v0.7.0 => github.com/celestiaorg/go-libp2p-pubsub v0.6.2-0.20220812132010-46b2a019f2f2
```

Comment (Bidon15) [reply go.mod:289]: And go mod tidy later 👍

**Review by renaynay — APPROVED** [review]
smorge - tACKed on testground

**Review by Bidon15 — APPROVED** [review]
🚢 
worked on testground with `0.7.0-rc1` app and core 1.3.3 v0.34.20

**Review by Wondertan — APPROVED** [review]
I am actually surprised that there are such small changes in here, and I am glad that the Tendermint team decided to keep some of the good changes and reverted some weird changes like `/websocket` one and `NewBlock` type changes. Also thanks for testing this on the Testground Viet. 

Just one question that I could find the answer my self, but asking here.

**Pull request merged**